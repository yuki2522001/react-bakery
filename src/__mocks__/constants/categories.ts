import { CategoryProps } from "@common-types/category";

const CATEGORY_MOCKING_LIST: CategoryProps[] = [
  {
    id: "1",
    name: "Cake"
  },
  {
    id: "2",
    name: "Breads"
  }
]

export { CATEGORY_MOCKING_LIST };
