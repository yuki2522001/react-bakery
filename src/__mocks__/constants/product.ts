import { Product } from "@common-types/product";
import url from "../../assets/images/products/black-forest-cake.jpg";

const PRODUCT_MOCKING: Product = {
  id: "1",
  categoryId: "1651924356334",
  name: "Cheese cake",
  price: 20,
  quantity: 5,
  description: "Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium.",
  images: [
    url
  ]
}

const PRODUCT_MOCKING_LIST: Product[] = [
  {
    id: "1",
    categoryId: "1651924356334",
    name: "Cheese cake",
    price: 20,
    quantity: 5,
    description: "Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium.",
    images: [
      "/src/assets/images/products/cheese-cake.jpg",
      "/src/assets/images/products/cheese-cake.jpg",
      "/src/assets/images/products/cheese-cake.jpg"
    ]
  },
  {
    id: "2",
    categoryId: "1651924356334",
    name: "Amaretti Cookie",
    price: 10,
    quantity: 5,
    description: "Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium.",
    images: [
      "/src/assets/images/products/cheese-cake.jpg",
      "/src/assets/images/products/cheese-cake.jpg",
      "/src/assets/images/products/cheese-cake.jpg"
    ]
  }
]

export { PRODUCT_MOCKING, PRODUCT_MOCKING_LIST };
