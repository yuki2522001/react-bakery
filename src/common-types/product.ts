import { CategoryProps } from "./category";

export interface Product {
  id: string;
  name: string;
  price: number;
  images: string[];
  quantity: number;
  categoryId: string;
  description: string;
  product?: {};
}

export interface ProductContext {
  products: Product[];
  dispatch: Function,
  searchValue: string;
  setSearchValue: Function;
  categories: CategoryProps[];
  setCategories: Function;
}