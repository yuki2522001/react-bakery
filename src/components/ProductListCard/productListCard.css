.productViewPage {
  display: flex;
}

.productViewPage__link {
  color: var(--text-master);
  text-decoration: none;
}

.productViewPage__image {
  width: 130px;
  height: 130px;
  border-radius: 50%;
  margin: 0 20px 30px 0;
}

.productViewPage__content {
  width: 420px;
}

.productViewPage__title {
  cursor: pointer;
  font-size: 16px;
}

.productViewPage__description {
  font-size: 14px;
  color: var(--text-secondary);
}

.productViewPage__price {
  color: var(--text-primary);
  font-weight: bold;
  margin: 0;
}