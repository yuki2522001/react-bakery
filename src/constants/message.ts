const SUCCESS_MSG = {
  MESSAGE_ADD_PRODUCT: "Product add successful",
  MESSAGE_UPDATE_PRODUCT: "Product update successful",
  MESSAGE_DELETE_PRODUCT: "Product delete successful",
};

const ERROR_MSG = {
  REQUIRED: "Field is required",
  NUMBER: "Enter a non-negative number"
}

export { SUCCESS_MSG, ERROR_MSG };
