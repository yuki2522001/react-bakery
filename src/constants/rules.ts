const RULES = {
  REQUIRED: "required",
  NUMBER: "number",
  NEGATIVE: "negative"
};

export { RULES }