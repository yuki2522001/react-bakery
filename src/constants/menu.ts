import { MenuType } from "../common-types/menu";

const MENU_LIST: MenuType[] = [
  {
    key: "about",
    label: "About",
  },
  {
    key: "shop",
    label: "Shop",
  },
  {
    key: "blog",
    label: "Blog",
  },
  {
    key: "faq",
    label: "FAQ",
  },
  {
    key: "contact",
    label: "Contact"
  }
];

export default MENU_LIST;
