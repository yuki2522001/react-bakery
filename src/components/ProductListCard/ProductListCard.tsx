import React from "react";
import { Link } from "react-router-dom";
import { Product } from "@common-types/product";
import Price from "@components/Price/Price";
import Text from "@components/Text/Text";
import Title from "@components/common/Title/Title";
import "./productListCard.css";

interface ProductCardProps {
  product: Product;
}

const ProductListCard: React.FC<ProductCardProps> = ({ product }) => {
  return (
    <div data-testid="product-list-card" className="productViewPage">
      <img className="productViewPage__image" src={product?.images[0]} />
      <div className="productViewPage__content">
        <Link className="productViewPage__link" to={`/products/${product.id}`}>
          <Title className="productViewPage__title" text={product.name} />
        </Link>
        <Text
          className="productViewPage__description"
          text={product.description}
        />
        <Price className="productViewPage__price" value={product.price} />
      </div>
    </div>
  );
};

export default ProductListCard;
