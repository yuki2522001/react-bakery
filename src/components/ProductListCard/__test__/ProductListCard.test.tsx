import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import ProductListCard from "../ProductListCard";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";
import { PRODUCT_MOCKING } from "@__mocks__/constants/product";

describe("Product list card component", () => {
  test("should render productListCard component", () => {
    const history = createMemoryHistory();
    const { getByTestId } = render (
      <Router location={history.location} navigator={history}>
        <ProductListCard product={PRODUCT_MOCKING} />
      </Router>
    );
    expect(getByTestId("product-list-card")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <Router location={history.location} navigator={history}>
        <ProductListCard product={PRODUCT_MOCKING} />
      </Router>);
    expect(asFragment()).toMatchSnapshot();
  });
})