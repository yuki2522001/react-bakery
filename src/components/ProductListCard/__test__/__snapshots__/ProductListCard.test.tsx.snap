// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product list card component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="productViewPage"
    data-testid="product-list-card"
  >
    <img
      class="productViewPage__image"
    />
    <div
      class="productViewPage__content"
    >
      <a
        class="productViewPage__link"
        href="/products/1"
      >
        <h2
          class="productViewPage__title"
          data-testid="title-test"
        >
          Cheese cake
        </h2>
      </a>
      <p
        class="productViewPage__description"
        data-testid="text"
      >
        Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium.
      </p>
      <p
        class="productViewPage__price"
        data-testid="price"
      >
        $20
      </p>
    </div>
  </div>
</DocumentFragment>
`;
